src/viewer.js

```diff
@@ -143,6 +143,14 @@ export class Viewer {
     /** @const @private {!Object<string, string>} */
     this.params_ = {};
 
+    /** @private {?function(*)} */
+    this.onVisibilityResolve_ = null;
```

Comment:
My point was simply that: if you already investing in implementing bunch of "document ready" promisable logic in the UserNotificationService - then you can just continue doing that. Or you can put all those promisable routines in each file separately, e.g. "Viewer.whenVisible" and "document-state:whenReady" - it's up to you. I do like the idea of having separate promisable calls in the viewer.js and document-state.js.